### PR TITLE
Added 'forget' feature to IPFS pinning section. 

### PR DESCRIPTION
**Pull request by leirbag95**
## Description
I have added a new feature to the IPFS Pinning section that allows users to "forget" a message. After archiving a message, users now have the option to unarchive or delete it. The "forget" option removes the message from the application. I have added the `forget-message` method to implement this feature.

## Changes Made
- Added a `forget-message` method
- Updated the UI to include a "Delete" button for removed messages
- Implemented the logic for permanently removing the message from the application when the "Delete" button is clicked
- In the same time I fixed  `total_used` undefined result issue
## Screenshots
Delete Button
Alert modal

## Note
Please let me know if you need any further information or clarification.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -71,6 +71,8 @@ module.exports = {
 
 
     // allow debugger during development only
-    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off'
+    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
+
+    "new-cap": 0
```
Comment (BjrInt) [inline .eslintrc.js:76]: Why did you add this rule instead of importing `shajs` constructor as `Shajs` for instance?

File: src/pages/IPFSPin.vue [hunk 2]
```diff
@@ -350,6 +371,70 @@ export default {
...
+    async putContent (message, content, inline_requested, api_server) {
+      let inline = inline_requested
+      if (inline) {
+        let serialized = JSON.stringify(content)
+        if (serialized.length > 150000) {
+          inline = false
+        } else {
+          message.item_type = 'inline'
+          message.item_content = serialized
+          // eslint-disable-next-line no-use-before-define
```
Comment (BjrInt) [inline src/pages/IPFSPin.vue:384]: Why put an eslint ignore, if you changed the rule anyway?

I would either:
- Use a capitalized constructor import
- OR disable the line
- OR (worst solution) change the eslint rule

But not the 3 solutions altogether.

File: src/pages/IPFSPin.vue [hunk 3]
```diff
@@ -228,10 +238,26 @@ export default {
...
+      selected_message: {
+        reason: 'None'
+      },
+      delete_prompt: false,
+      delete_field: ''
     }
   },
   methods: {
+    delete_rules (val) {
+      return new Promise((resolve, reject) => {
```
Comment (BjrInt) [inline src/pages/IPFSPin.vue:251]: What is this Promise actually doing? Why the timeout?

File: src/pages/IPFSPin.vue [hunk 4]
```diff
@@ -350,6 +371,70 @@ export default {
           path: this.cid_name
         })
       }
+    },
+    async putContent (message, content, inline_requested, api_server) {
+      let inline = inline_requested
+      if (inline) {
+        let serialized = JSON.stringify(content)
+        if (serialized.length > 150000) {
```
Comment (BjrInt) [inline src/pages/IPFSPin.vue:379]: The threshold is in bytes, not in character length. I think 150k characters (up to 600K bytes in the worst UTF-8 case) is a bit too high and the message will be rejected by the node. 
Did you test that your limit is actually working?

File: src/pages/IPFSPin.vue [hunk 5]
```diff
@@ -350,6 +371,70 @@ export default {
...
+          message.item_hash = new shajs.sha256().update(serialized).digest('hex') // eslint-disable-line no-use-before-define
+        }
+      }
+    },
+    async send (message) {
+      if (this.account.type === 'ETH') {
+        message = await ethereum.sign(this.account, message)
+      } else if (this.account.type === 'SOL') {
+        message = await solana.sign(this.account, message)
+      }
```
Comment (BjrInt) [inline src/pages/IPFSPin.vue:394]: Could this could be abstracted in the store, so you would always use the right signer instead of having a conditional switch for every message that you create?

**Review by BjrInt — CHANGES_REQUESTED** [review]

File: src/pages/IPFSPin.vue [hunk 6]
```diff
@@ -152,11 +159,12 @@
...
 import 'vue2-dropzone/dist/vue2Dropzone.min.css'
 import { mapState } from 'vuex'
-import { store, aggregates } from 'aleph-js'
+import { store, broadcast, aggregates, ethereum, solana } from 'aleph-js'
 import { format, copyToClipboard } from 'quasar'
 const { humanStorageSize } = format
 
 import IPFS from 'ipfs'
+const shajs = require('sha.js')
 const isIPFS = require('is-ipfs')
```
Comment (BjrInt) [inline src/pages/IPFSPin.vue:168]: Please stick to a style of imports ESModules OR CommonJS.

File: src/pages/IPFSPin.vue (hunk 3 above)
Comment (leirbag95) [reply src/pages/IPFSPin.vue:251]: Each time the user enters a character in the "DELETE" field a loading animation is played, this animation lasts 0.5 seconds

File: src/pages/IPFSPin.vue (hunk 4 above)
Comment (leirbag95) [reply src/pages/IPFSPin.vue:379]: I didn't write this code, I copied it directly from aleph-js (https://github.com/aleph-im/aleph-js/blob/359ecae06ac5580f07d5790a43ea3a119b607a3d/src/api/create.js#L14) and removed the storage part.
But I can decrease the max length

File: .eslintrc.js (hunk 1 above)
Comment (leirbag95) [reply .eslintrc.js]: Done, I import sha256 as Sha256 now

```{js}
import { sha256 as Sha256 } from 'sha.js'
```

File: src/pages/IPFSPin.vue (hunk 6 above)
Comment (leirbag95) [reply src/pages/IPFSPin.vue]: Done

File: src/pages/IPFSPin.vue (hunk 3 above)
Comment (BjrInt) [reply src/pages/IPFSPin.vue:252]: I don't think that animation is necessary, just disabling the "delete" button seems enough. 
There is a "loading" animation, but this gives confusing feedback, as there is nothing to load really. 

Looking at how Github handles repo deletion for instance, there is no loading animation.

Comment (leirbag95) [reply src/pages/IPFSPin.vue]: I removed the animation

**Review by leirbag95 — COMMENTED** [review]
change reviewed

**Comment by BjrInt** [comment]
Still one more thing, to fix, when I succesfully delete an item, the DELETE field is already filled.
[Screencast from 27-01-23 13:27:07.webm](https://user-images.githubusercontent.com/26065817/215087904-b0a73221-ad49-45aa-a657-f114c6a21f56.webm)

**Comment by leirbag95** [comment]
> Deployed (not pinned) on https://ipfs-2.aleph.im/ipfs/QmZbB5X9nNDfKBUaw1numTt67A3kt4xaR2Huk7uG6ZG3ha

@BjrInt This is how I fixed the DELETE prompt issue from `IPFSPin.vue`

```
...
closePopup () {
      this.delete_field = ''
...

```

**Review by BjrInt — APPROVED** [review]
LGTM
